src/java/org/apache/cassandra/service/disk/usage/DiskUsageMonitor.java

```diff
@@ -79,14 +80,22 @@ public void start(Consumer<DiskUsageState> notifier)
     {
         // start the scheduler regardless guardrail is enabled, so we can enable it later without a restart
         ScheduledExecutors.scheduledTasks.scheduleAtFixedRate(() -> {
-
-            if (!Guardrails.localDataDiskUsage.enabled(null))
-                return;
-
-            updateLocalState(getDiskUsage(), notifier);
+            boolean currentEnabledStatus = Guardrails.localDataDiskUsage.enabled(null);
+            boolean oldEnabledStatus = enabled;
+            enabled = currentEnabledStatus;
+            if (!currentEnabledStatus && oldEnabledStatus)
+                onDiskUsageGuardrailDisabled(notifier);
+            else
+                updateLocalState(getDiskUsage(), notifier);
```

Comment:
Optional, may be consider extracting the if condition into a variable for better readability?


```suggestion
boolean currentEnabled = Guardrails.localDataDiskUsage.enabled(null);
boolean oldEnabled = enabled;
boolean isDisabled = !currentEnabled && oldEnabled;
enabled = currentEnabled;
if (isDisabled)
     onDiskUsageGuardrailDisabled(notifier);
else
      updateLocalState(getDiskUsage(), notifier);
```